tests/unit/src/lib/validator.test.js

```diff
@@ -0,0 +1,116 @@
+const { isValidNino } = require("../../../../src/lib/validator");
+
+const good_ninos = [
+  "AA283902A",
+  "BA283902A",
+  "CA283902A",
+  "EA283902A",
+  "GA283902A",
+  "HA283902A",
+  "JA283902A",
+  "KA283902A",
+  "LA283902A",
+  "MA283902A",
+  "NA283902A",
+  "OA283902A",
+  "PA283902A",
+  "RA283902A",
+  "SA283902A",
+  "TA283902A",
+  "WA283902A",
+  "XA283902A",
+  "YA283902A",
+  "ZA283902A",
+  "AA283902A",
+  "AB283902A",
+  "AC283902A",
+  "AE283902A",
+  "AG283902A",
+  "AH283902A",
+  "AJ283902A",
+  "AK283902A",
+  "AL283902A",
+  "AM283902A",
+  "AN283902A",
+  "AP283902A",
+  "AR283902A",
+  "AS283902A",
+  "AT283902A",
+  "AW283902A",
+  "AX283902A",
+  "AY283902A",
+  "AZ283902A",
+  "AA283902A",
+  "AA283902B",
+  "AA283902C",
+  "AA283902D",
+];
+const bad_ninos = [
```

Comment:
Given the regex groups above, a potentially useful refactoring would be to break this up into the groups above, so we're clear about which bits we're testing. Otherwise it's a block of data being validated with no context.